src/mcp/tools/dataconnect/info.ts

```diff
@@ -0,0 +1,149 @@
+    const serviceInfos = new Map<string, CombinedServiceInfo>();
+
+    for (const l of localServiceInfos) {
+      serviceInfos.set(
+        `locations/${l.dataConnectYaml.location}/services/${l.dataConnectYaml.serviceId}`,
+        { local: l },
+      );
+    }
+
+    if (projectId) {
+      try {
+        const [services, schemas, connectors] = await Promise.all([
+          client.listAllServices(projectId),
+          client.listSchemas(`projects/${projectId}/locations/-/services/-`),
+          client.listConnectors(`projects/${projectId}/locations/-/services/-`),
+        ]);
+        console.log(services, schemas, connectors);
+        for (const s of services) {
+          const k = s.name.split("/").slice(2, 6).join("/");
+          const st = serviceInfos.get(k) || {};
+          st.deployed = st.deployed || {};
+          st.deployed.service = s;
+          serviceInfos.set(k, st);
+        }
+        for (const s of schemas) {
+          const k = s.name.split("/").slice(2, 6).join("/");
+          const st = serviceInfos.get(k) || {};
+          st.deployed = st.deployed || {};
+          st.deployed.schemas = st.deployed.schemas || [];
+          st.deployed.schemas.push(s);
+          serviceInfos.set(k, st);
+        }
+        for (const s of connectors) {
+          const k = s.name.split("/").slice(2, 6).join("/");
+          const st = serviceInfos.get(k) || {};
+          st.deployed = st.deployed || {};
+          st.deployed.connectors = st.deployed.connectors || [];
+          st.deployed.connectors.push(s);
+          serviceInfos.set(k, st);
+        }
+      } catch (e: any) {
+        logger.debug("cannot fetch dataconnect resources in the backend", e);
+      }
+    }
+
+    const localServices = Array.from(serviceInfos.values()).filter((s) => s.local);
+    const remoteOnlyServices = Array.from(serviceInfos.values()).filter((s) => !s.local);
+
+    const output: string[] = [];
+
+    function includeDeployedServiceInfo(deployed: DeployServiceInfo): void {
+      if (deployed.schemas?.length) {
+        output.push(`### Schemas`);
+        for (const s of deployed.schemas) {
+          clearCCFEFields(s);
+          output.push(dump(s));
+        }
+      }
+      if (deployed.connectors?.length) {
+        output.push(`### Connectors`);
+        for (const c of deployed.connectors) {
+          clearCCFEFields(c);
+          output.push(dump(c));
+        }
+      }
+    }
+
+    if (localServices.length) {
+      output.push(`# Local Data Connect Sources`);
+      for (const s of localServices) {
+        const local = s.local!;
+        output.push(dump(local.dataConnectYaml));
+        const schemaDir = path.join(local.sourceDirectory, local.dataConnectYaml.schema.source);
+        output.push(`You can find all of schema sources under ${schemaDir}/`);
+        if (s.deployed) {
+          output.push(`It's already deployed in the backend:\n`);
+          includeDeployedServiceInfo(s.deployed);
```

Comment:
This may product empty output, as shown in your screenshot in the PR description